Certs (#209)
kyma-project/btp-manager: docs/certs.md

![Certification management diagram](./assets/certs.svg)

## Certification management

> BTP Manager maintain two secrets `ca-server-cert` and `webhook-server-cert` which are used to allow communication within BTP Operator webhooks, and in result allow to create resources like ServiceInstances and ServiceBindings

Reviewer: ```suggestion
BTP Manager maintains two secrets, `ca-server-cert` and `webhook-server-cert`, which are used to allow communication within BTP Operator webhooks and thus allow the creation of resources like ServiceInstances and ServiceBindings.
```

> During provisioning, first there is created `ca-server-cert` which is self sign CA certificate, then based on that, application create signed cert `webhook-server-cert`, which is mounted under the deployment.

Reviewer: ```suggestion
During provisioning, at first, `ca-server-cert` is created. It is a self-sign CA certificate. Then based on that, the application creates a signed cert, `webhook-server-cert`, which is mounted under the deployment.
```

> Webhooks have CABundle field which is set to the content of ca-server-cert and this field is managed by BTP Manager.

Reviewer: ```suggestion
Webhooks have a CABundle field set to the content of `ca-server-cert,` and BTP Manager manages this field.
```

> Webhooks have CABundle field which is set to the content of ca-server-cert and this field is managed by BTP Manager.

Reviewer: You use `caBundle`, `CA bundle`, `ca bundle` and `CABundle`. If they're not three different things, why don't you use the same spelling in the whole doc?

> The `ca-server-cert`, `webhook-server-cert`, and webhooks caBundles are keep in sync by using reconcilation mechanism, which mean every manual change by user in this resources will trigger automatically regeneration of all three resources.

Reviewer: ```suggestion
The `ca-server-cert`, `webhook-server-cert`, and webhooks' caBundles are kept in sync by using the reconciliation mechanism, which means every manual change in these resources by the user will automatically trigger regeneration of all three resources.
```

> BTP Manager maintain resources by creating, deleting and updating actions during reconcilation. The goal is to keep `ca-server-cert`, `webhook-servert-cert` and webhooks ca bundles in sync, all the time.

Reviewer: ```suggestion
BTP Manager maintains resources by creating, deleting, and updating actions during reconciliation. The goal is to keep `ca-server-cert`, `webhook-server-cert`, and webhooks' ca bundles in sync all the time.
```

> Scheduled reconcilation check also expiration dates of certificates and if it detected that certificate expires soon, it regenerates it in advanced, to keep things smoothly.

Reviewer: ```suggestion
Scheduled reconciliation also checks expiration dates of certificates, and if it detects that a certificate expires soon, it regenerates it in advance so that the processes run smoothly.
```

---
Certification management
---

![Certification management diagram](./assets/certs.svg)

> 

Reviewer: Some more comments on the diagram:
1. The font color in the third diamond from the left seems inconsistent with the rest of them.
2. The questions in the last block (Expiration) are incorrect. Remove `does` from the last two diamonds. How about rephrasing them to `cert from (...) secret expires within a given time?`?
3. Remove `is` from the fourth diamond and `no` from the first two diamonds.
4. You use `CA bundle` in the diagram and `caBundle` in the description. Shouldn't they be the same?
5. In the top left-hand corner of the diagram there's `self signed`. Change it to `self-signed`.
6. If there are questions, there should be some answers. How about adding yes/no arrows?
